src/System.Diagnostics.Process/src/System/Diagnostics/ProcessStartInfo.cs

```diff
@@ -79,16 +89,47 @@ public bool CreateNoWindow
             set { _createNoWindow = value; }
         }
 
+        public StringDictionary EnvironmentVariables 
+        {
+            get 
+            {
+                                
+                if (_environmentVariables == null)
+                {
+#if PLATFORM_UNIX
+                    _environmentVariables = new StringDictionaryWithComparer(StringComparer.Ordinal);
+#else
+                    _environmentVariables = new StringDictionaryWithComparer(StringComparer.OrdinalIgnoreCase);
+#endif // PLATFORM_UNIX
+
+                    IEnumerator<KeyValuePair<string,string>> e = Environment.GetEnumerator();
+                    try
+                    {
+                        while (e.MoveNext())
+                        {
+                            _environmentVariables.Add((string)e.Current.Key, (string)e.Current.Value);
+                        }
+                    }
+                    finally
+                    {
+                        (e as IDisposable)?.Dispose();
+                    }
+                
+                }
+                return _environmentVariables;
+            }
+        }
+
         public IDictionary<string, string> Environment
         {
             get
             {
-                if (_environmentVariables == null)
+                if (_environment == null)
```

Comment:
Would this workaround work better? 

https://github.com/safern/corefx/blob/2ad2af1f7e472a5d4cc6508a3abda6d1a0783b19/src/System.Diagnostics.Process/src/System/Diagnostics/ProcessStartInfo.cs#L91

Only creating a StringDictionaryWithComparer with the Environment dictionary as its contents.

@stephentoub